source/simplex/fe_lib.cc

```diff
@@ -1187,7 +1203,274 @@ namespace Simplex
+                lump_polys.push_back(std::move(p));
+
+                // Sanity check:
+#ifdef DEBUG
+              BarycentricPolynomial<dim> unity;
+              for (const auto &p : lump_polys)
+                unity = unity + p;
+
+              Point<dim> test;
+              for (unsigned int d = 0; d < dim; ++d)
+                test[d] = 2.0;
+              Assert(std::abs(unity.value(test) - 1.0) < 1e-10,
+                     ExcInternalError());
+#endif
+
+              return BarycentricPolynomials<dim>(lump_polys);
+            }
+          default:
+            Assert(degree < 3, ExcNotImplemented());
+        }
+
+      Assert(degree < 3, ExcNotImplemented());
+      // bogus return to placate compilers
+      return BarycentricPolynomials<dim>::get_fe_p_basis(degree);
+    }
+
+
+
+    template <int dim>
+    FiniteElementData<dim>
+    get_fe_data(const unsigned int degree)
+    {
+      // It's not efficient, but delegate computation of the degree of the
+      // finite element (which is different from the input argument) to the
+      // basis.
+      const auto polys = get_basis<dim>(degree);
+      return FiniteElementData<dim>(get_dpo_vector<dim>(degree),
+                                    ReferenceCell::Type::get_simplex<dim>(),
+                                    1, // n_components
+                                    polys.degree(),
+                                    FiniteElementData<dim>::H1);
+    }
+  } // namespace FE_P_BubblesImplementation
+
+
+
+  template <int dim, int spacedim>
+  FE_P_Bubbles<dim, spacedim>::FE_P_Bubbles(const unsigned int degree)
+    : dealii::FE_Poly<dim, spacedim>(
+        FE_P_BubblesImplementation::get_basis<dim>(degree),
+        FE_P_BubblesImplementation::get_fe_data<dim>(degree),
+        std::vector<bool>(
+          FE_P_BubblesImplementation::get_fe_data<dim>(degree).dofs_per_cell,
+          true),
+        std::vector<ComponentMask>(
+          FE_P_BubblesImplementation::get_fe_data<dim>(degree).dofs_per_cell,
+          std::vector<bool>(1, true)))
+    , approximation_degree(degree)
+  {
+    this->unit_support_points =
+      FE_P_BubblesImplementation::unit_support_points<dim>(degree);
+
+    // TODO
+    // this->unit_face_support_points =
+    //   unit_face_support_points_fe_poly<dim>(degree);
+  }
+
+
+
+  template <int dim, int spacedim>
+  std::string
+  FE_P_Bubbles<dim, spacedim>::get_name() const
+  {
+    return "Simplex::FE_P_Bubbles<" + Utilities::dim_string(dim, spacedim) +
+           ">" + "(" + std::to_string(approximation_degree) + ")";
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  FE_P_Bubbles<dim, spacedim>::
+    convert_generalized_support_point_values_to_dof_values(
+      const std::vector<Vector<double>> &support_point_values,
+      std::vector<double> &              nodal_values) const
+  {
+    AssertDimension(support_point_values.size(),
+                    this->get_unit_support_points().size());
```

Comment:
Is this check actually needed?

Reply:
I think so - `support_point_values` is a function input.